### DISCUSSION TITLE
Minimum value for .spec.interval on Bucket

### DISCUSSION THREAD

**Discussion by leonaves** [fluxcd/flux2#4702] [Q&A]
As title says, is there a minimum value for .spec.interval on Bucket in flux itself? or if not are there platform-specific minimums? Can I fetch from S3 every 2 seconds for example?

**Comment by stefanprodan** [comment]
You can set any interval, but S3 will rate limit and if you abuse it, your cluster IPs will be blocked. Setting a value under 1m is not something that I would do, besides DDOS-ing your cloud provider you'll also increase the CPU and network usage of Flux. If you are running a Minio sever in same network as the cluster, in that case I think you could lower the interval under 1m if you really need to.

**Reply by leonaves** [reply]
Hmm okay interesting thank you, we're essentially looking to dynamically spin up workloads within a few seconds (i.e. we don't want a 1 minute delay) and we may have thousands of workloads idle at any one time. We're probably not spinning up _thousands_ in a few seconds but of the thousands of idle ones we have we want to be able to spin some up at a moments notice, so we were hoping to use flux for this, i.e. if we point it at a bucket with thousands of deployment specs, we want it to notice replicas going from 0 to 1 on any given one within a few seconds.

**Reply by stefanprodan** [reply]
You should be using a Flux webhook receiver to trigger a reconciliation right after you update the bucket. Lots of users do this from CI. See https://fluxcd.io/flux/guides/webhook-receivers/

**Reply by leonaves** [reply]
Would this cause problems if the bucket updated lets say 50-100 times a second?

**Reply by stefanprodan** [reply]
I don’t think it would cause much issues, no matter how much you spam the webhook receiver, Flux will reconcile the new workloads once those in-flight are done.
